vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java

```diff
@@ -180,6 +195,72 @@
                     .where(ACTIVE_OPERATION.WORKFLOW_RUN_ID.eq(WORKFLOW_RUN.ID)))));
   }
 
+  private static Field<?> createQuery(VersionPolicy policy, Set<String> allowedTypes) {
+    switch (policy) {
+      case ALL:
+        {
+          var condition = EXTERNAL_ID.ID.eq(EXTERNAL_ID_VERSION.EXTERNAL_ID_ID);
+          if (allowedTypes != null) {
+            condition = condition.and(EXTERNAL_ID_VERSION.KEY.in(allowedTypes));
+          }
+          final var externalIdVersionAlias = EXTERNAL_ID_VERSION.as("externalIdVersionInner");
+          final var table =
+              DSL.selectDistinct(EXTERNAL_ID_VERSION.KEY.as("desired_key"))
+                  .from(EXTERNAL_ID_VERSION)
+                  .where(condition.and(EXTERNAL_ID_VERSION.EXTERNAL_ID_ID.eq(EXTERNAL_ID.ID)))
```

Comment:
the additional condition here is duplicated from above (L202)

Reply:
Wow. Yes. I didn't see this and I don't know how you did. Fixed.